src/encoded/batch_download.py

```diff
@@ -216,13 +247,123 @@ def peak_metadata(context, request):
     )
 
 
+def _get_annotation_metadata(request, search_path):
+    """
+    Get annotation metadata.
+
+        :param request: Pyramid's request.
+        :param search_path: Search url.
+    """
+    param_list = parse_qs(request.matchdict['search_params'])
+    param_list['limit'] = ['all']
+    # Handle metadata.tsv lines from cart-generated files.txt.
+    # TODO: When rewriting this file, refactor this duplicated cart code.
+    cart_uuids = param_list.get('cart', [])
+    if cart_uuids:
+        cart_uuid = cart_uuids.pop()
+        del param_list['cart']
+        try:
+            cart = request.embed(cart_uuid, '@@object')
+        except KeyError:
+            pass
+        else:
+            if cart.get('elements'):
+                param_list['@id'] = cart['elements']
+    else:
+        try:
+            elements = request.json.get('elements')
+        except ValueError:
+            pass
+        else:
+            param_list['@id'] = elements
+    path = '{}?{}'.format(search_path, urlencode(param_list, True))
```

Comment:
I think you need to add limit=all here if there are > 25 results.

Reply:
@hitz I assume you mean param_list['all']. If correct, this is a chicken-egg problems because param_list is needed to get results so results cannot be made dependent on param_list.

Reply:
oh I see it was added here: C/encoded/blob/1ae7afda60c56b732a3512df1a4e0d677d771916/src/encoded/batch_download.py#L258, I missed it when you moved it to the function.